src/Documentation/SidebarMenu/SidebarMenuHelper.js

```diff
@@ -0,0 +1,161 @@
+    let filename = SidebarMenuHelper.extractFilename(file)
+    let path = SidebarMenuHelper.getFullPath(folder, filename)
+    arr[path] = startCase(
+      SidebarMenuHelper.removeExtensionFromFileName(filename)
+    )
+  }
+
+  static isString = obj => typeof obj === 'string'
+
+  static transformAbsoluteToDocRelatedPath = path =>
+    path
+      .join(PATH_SEPARATOR)
+      .replace(new RegExp(PATH_TO_DOC, 'i'), '')
+      .split(PATH_SEPARATOR)
+      .filter(_ => _)
+
+  static combineToPath = subPaths => [].concat(subPaths).join(PATH_SEPARATOR)
+
+  static getZeroFile = arr => {
+    const firstItem = arr[0]
+    const { files, indexFile } = firstItem
+    return (
+      (files && SidebarMenuHelper.getZeroFile(files)) || indexFile || firstItem
+    )
+  }
+
+  static findFileByName = (item, find) => {
+    let file = null
+    if (
+      SidebarMenuHelper.removeExtensionFromFileName(item.indexFile || item) ===
+      find
+    ) {
+      file = item
+    } else if (kebabCase(item.name || '') === find) {
+      file = item.files[0]
+    }
+    return file
+  }
+
+  static getFile = (arr, find, indexPush, setFile) => {
+    arr.forEach((item, index) => {
+      let newfile = SidebarMenuHelper.findFileByName(item, find)
+      if (newfile) {
+        indexPush(index)
+        setFile(newfile)
+      } else if (item.files) {
+        SidebarMenuHelper.getFile(item.files, find, indexPush, setFile)
+      }
+    })
+  }
+
+  static getFileFromUrl = path => {
+    let indexes = [],
+      file = SidebarMenuHelper.getZeroFile(sidebar),
+      newpath = SidebarMenuHelper.transformAbsoluteToDocRelatedPath(path)
+    newpath.forEach(part => {
+      SidebarMenuHelper.getFile(
+        sidebar,
+        part,
+        i => {
+          indexes.push(i)
+        },
+        f => {
+          file = f
+        }
+      )
+    })
+    return {
+      file: file,
+      indexes: indexes
+    }
+  }
+
+  static getName = (labels = null, folder = null, indexFile = null, names) => {
+    let name
+    if (labels && labels[indexFile]) {
+      name = labels[indexFile]
+    } else {
+      let path = SidebarMenuHelper.getFullPath(folder, indexFile)
+      name = names[path]
+    }
+    return name
+  }
+
+  static getNamesArr = json => {
+    let arr = {}
+    json.map(section => {
+      section.files.map(file => {
+        SidebarMenuHelper.fillFilesArray(section, file, arr)
+        if (SidebarMenuHelper.hasChildrenFiles(file)) {
+          file.files.map(subFile => {
+            SidebarMenuHelper.fillFilesArray(section, subFile, arr)
+          })
+        }
+      })
+    })
+    return arr
+  }
+
+  static convertToBooleanString(cond) {
+    return (!!cond).toString()
+  }
+
+  static filesContains(array, folder, currentFile) {
+    let flag = false
+    array.forEach(elem => {
+      let path = SidebarMenuHelper.getFullPath(folder, elem)
+      if (path === currentFile) {
+        flag = true
```

Comment:
return true

Reply:
during the passage of the array, in the end we need only the last value